ParkingLotTest/ParkingLotTest.cs

```diff
@@ -0,0 +1,121 @@
+            ParkingBoy parkingBoy = new ParkingBoy(parkingLots);
+            ParkingTicket parkingTicket = parkingBoy.ParkingCar(car, out positionErrorMessage);
+            Car actual = parkingBoy.FetchCar(parkingTicket, out errorMessage);
+
+            // then
+            Assert.Equal(car, actual);
+        }
+
+        [Fact]
+        public void ShouldFetchACarUsingCorrespondingParkingTicket()
+        {
+            // given
+            Car car1 = new Car("CARA111");
+            Car car2 = new Car("CARA112");
+
+            // when
+            string errorMessage, positionErrorMessage1, positionErrorMessage2;
+            ParkingLot parkingLot1 = new ParkingLot("A");
+            ParkingLot parkingLot2 = new ParkingLot("B");
+            List<ParkingLot> parkingLots = new List<ParkingLot>();
+            parkingLots.Add(parkingLot1);
+            parkingLots.Add(parkingLot2);
+            ParkingBoy parkingBoy = new ParkingBoy(parkingLots);
+            ParkingTicket parkingTicket1 = parkingBoy.ParkingCar(car1, out positionErrorMessage1);
+            ParkingTicket parkingTicket2 = parkingBoy.ParkingCar(car2, out positionErrorMessage2);
+            Car actual = parkingBoy.FetchCar(parkingTicket2, out errorMessage);
+
+            // then
+            Assert.Equal(car2, actual);
+        }
+
+        [Fact]
+        public void ShouldGiveErrorMessageWhenTicketIsNull()
+        {
+            // given
+            Car car = new Car("CARA111");
+
+            // when
+            string errorMessage, positionErrorMessage;
+            ParkingLot parkingLot1 = new ParkingLot("A");
+            ParkingLot parkingLot2 = new ParkingLot("B");
+            List<ParkingLot> parkingLots = new List<ParkingLot>();
+            parkingLots.Add(parkingLot1);
+            parkingLots.Add(parkingLot2);
+            ParkingBoy parkingBoy = new ParkingBoy(parkingLots);
+            ParkingTicket parkingTicket = parkingBoy.ParkingCar(car, out positionErrorMessage);
+            Car actual = parkingBoy.FetchCar(null, out errorMessage);
+
+            // then
+            Assert.Equal("Please provide your parking ticket.", errorMessage);
+        }
+
+/*
+        [Fact]
+        public void ShouldGiveErrorMessageWhenTicketIsUsed()
+        //ShouldGiveErrorMessageWhenGiveWrongTicket()
+        {
+            // given
+            Car car1 = new Car("CARA111");
+            Car car2 = new Car("CARA112");
+
+            // when
+            string errorMessage1, errorMessage2;
+            ParkingBoy parkingBoy = new ParkingBoy();
+            ParkingTicket parkingTicket1 = parkingBoy.ParkingCar(car1);
+            ParkingTicket parkingTicket2 = parkingBoy.ParkingCar(car2);
+            parkingBoy.FetchCar(parkingTicket1, out errorMessage1);
+            parkingBoy.FetchCar(parkingTicket1, out errorMessage2);
+
+            // then
+            Assert.Equal("Unrecognized parking ticket.", errorMessage2);
+        }
+*/
+    }
+}
```

Comment:
duplication in different test methods can be solved by IDisposable interface
https://xunit.github.io/docs/shared-context.html#class-fixture